frontend/packages/topology/src/filters/FilterProvider.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { useUserSettings } from '@console/shared';
+import { getDefaultAppliedFilters } from '../redux/reducer';
+import { DEFAULT_TOPOLOGY_FILTERS } from './const';
+import { getAppliedFilters } from '../redux/action';
+import { DisplayFilters } from '../topology-types';
+
+const TOPOLOGY_DISPLAY_FILTERS_CONFIG_MAP_KEY = `devcosole.topology.filters`;
+
+const getTopologyFilters = (appliedFilters) => {
+  const filters = [...DEFAULT_TOPOLOGY_FILTERS];
+  filters.forEach((filter) => {
+    if (appliedFilters[filter.id] !== undefined) {
+      filter.value = appliedFilters[filter.id];
+    }
+  });
+
+  return filters;
+};
+
+type AppliedFilters = { [filterKey: string]: boolean };
+type SetTopologyFilters = (filters: DisplayFilters) => void;
+
+const useFilterContextValues = (): [
+  DisplayFilters,
+  AppliedFilters,
+  boolean,
+  SetTopologyFilters,
+] => {
+  const [appliedFilters, setAppliedFilters, appliedFiltersLoaded] = useUserSettings(
+    TOPOLOGY_DISPLAY_FILTERS_CONFIG_MAP_KEY,
+    getDefaultAppliedFilters(),
+  );
+  const [filtersLoaded, setFiltersLoaded] = React.useState<boolean>(false);
```

Comment:
Do we need this useState , can't we rely on `appliedFiltersLoaded`?

Reply:
Yes, this is needed because filters depend on the data which is coming from user settings(appliedFilters). so once the user settings is loaded we need to update the filters as well and notify the component that all of the data is loaded.